include/oneapi/dpl/pstl/hetero/dpcpp/sycl_iterator.h

```diff
@@ -233,6 +233,11 @@ __internal::sycl_iterator<access_mode::discard_read_write, T, Allocator> end(syc
     return __internal::sycl_iterator<access_mode::discard_read_write, T, Allocator>{buf,
                                                                                     __dpl_sycl::__get_buffer_size(buf)};
 }
+
+template <typename ValueType>
+constexpr static bool usm_allocated_vector_iterators_supported_v =
```

Comment:
1) I guess the traits doesn't depend on `ValueType`.
2) As an idea - to make such public trait more general, for example,
```
template <typename Container>
constexpr static bool is_usm_container_supported = ...
```

Reply:
Sadly, I believe it can depend on the `ValueType`, for example `std::vector<bool>`.  GNU c++ standard library implementation normally includes allocator type in the iterator, but not for `std::vector<bool>` with the bitfield "optimization".